tracegen/main.go

```diff
@@ -47,18 +49,27 @@ func main() {
 		zap.AddCallerSkip(3),
 	))
 
-	expOptions := []otlptracegrpc.Option{
+	grpcExpOpt := []otlptracegrpc.Option{
 		otlptracegrpc.WithEndpoint(cfg.Endpoint),
 		otlptracegrpc.WithDialOption(
 			grpc.WithBlock(),
 		),
 	}
 
+	httpExpOpt := []otlptracehttp.Option{
+		otlptracehttp.WithEndpoint(cfg.Endpoint),
+	}
+
 	if cfg.Insecure {
-		expOptions = append(expOptions, otlptracegrpc.WithInsecure())
+		grpcExpOpt = append(grpcExpOpt, otlptracegrpc.WithInsecure())
+		httpExpOpt = append(httpExpOpt, otlptracehttp.WithInsecure())
+	}
+
+	var exp *otlptrace.Exporter
+	if exp, err = otlptracegrpc.New(context.Background(), grpcExpOpt...); cfg.HttpExp {
+		exp, err = otlptracehttp.New(context.Background(), httpExpOpt...)
 	}
 
-	exp, err := otlptracegrpc.New(context.Background(), expOptions...)
 	if err != nil {
 		logger.Error("failed to obtain OTLP exporter", zap.Error(err))
```

Comment:
This should probably log which exporter was used? Perhaps it would be easier to move this entire iferr to the conditional block above.

Reply:
Hmm how about adding a log in case of HTTP exporter instead?
`var exp *otlptrace.Exporter
	if exp, err = otlptracegrpc.New(context.Background(), grpcExpOpt...); cfg.UseHTTP {
		logger.Info("starting HTTP exporter")
		exp, err = otlptracehttp.New(context.Background(), httpExpOpt...)
	}`

Reply:
Positive confirmation is always good. Add it for both gRPC and HTTP, please.

Reply:
Added positive info for both cases in order to avoid nested iferr - what do you think?

Reply:
Looks good!